sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -3,21 +3,56 @@
 # Licensed under the MIT License. See LICENSE.txt in the project root for
 # license information.
 # -------------------------------------------------------------------------
-from typing import Any, NamedTuple, Optional
+from typing import Any, NamedTuple, Optional, TypedDict
 from typing_extensions import Protocol, runtime_checkable
 
 
 class AccessToken(NamedTuple):
     """Represents an OAuth access token."""
 
     token: str
+    """The token string."""
     expires_on: int
-    refresh_on: Optional[int] = None
+    """The token's expiration time in Unix time."""
 
 
-AccessToken.token.__doc__ = """The token string."""
-AccessToken.expires_on.__doc__ = """The token's expiration time in Unix time."""
-AccessToken.refresh_on.__doc__ = """When the token should be refreshed in Unix time."""
+class AccessTokenInfo:
+    """Information about an OAuth access token.
+
+    This class is an alternative to `AccessToken` which provides additional information about the token.
+
+    :param str token: The token string.
+    :param int expires_on: The token's expiration time in Unix time.
+    :keyword int refresh_on: Specifies the time, in Unix time, when the cached token should be proactively
+        refreshed. Optional.
+    """
+
+    token: str
+    """The token string."""
+    expires_on: int
+    """The token's expiration time in Unix time."""
+    refresh_on: Optional[int]
+    """Specifies the time, in Unix time, when the cached token should be proactively refreshed. Optional."""
+
+    def __init__(self, token: str, expires_on: int, *, refresh_on: Optional[int] = None) -> None:
+        self.token = token
+        self.expires_on = expires_on
+        self.refresh_on = refresh_on
```

Comment:
Let's also add token_type which is needed for pop.

Reply:
Does it really needed to be added in this PR? PoP is a whole new set of APIs/features where this can be added in that PR.

Reply:
Discussed with Scott, we had not finalized the design yet so we may hold off on the change in policies.

the token_type is relatively a safe change to unblock the pop token.

We still need to confirm the property name and type when Chris is back.

Reply:
Added in latest changeset.